src/content/doc-surrealql/datamodel/idioms.mdx

```diff
@@ -0,0 +1,601 @@
+
+```surql
+SELECT name, spouse?.name AS spouse_name FROM person;
+```
+
+This idiom will return `null` for `spouse_name` if the `spouse` field is not present.
+
+## Combining Idiom Parts
+
+Idioms can combine multiple parts to navigate complex data structures seamlessly.
+
+### Complex Example
+
+Suppose we have the following data:
+
+```surql title="Create a new person record"
+CREATE person:5 CONTENT {
+    name: "Eve",
+    friends: [
+        {
+            id: "person:6",
+            name: "Frank",
+            age: 25
+        },
+        {
+            id: "person:7",
+            name: "Grace",
+            age: 19
+        },
+        {
+            id: "person:8",
+            name: "Heidi",
+            age: 17
+        }
+    ]
+};
+```
+
+```json title="Response"
+{
+  "id": "person:5",
+  "name": "Eve",
+  "friends": [
+    {
+      "id": "person:6",
+      "name": "Frank",
+      "age": 25
+    },
+    {
+      "id": "person:7",
+      "name": "Grace",
+      "age": 19
+    },
+    {
+      "id": "person:8",
+      "name": "Heidi",
+      "age": 17
+    }
+  ]
+}
+```
+
+To get the names of friends who are over 18:
+
+```surql
+SELECT friends[WHERE age > 18].name FROM person WHERE id = r'person:5';
+```
+
+```json title="Response"
+[
+	{
+		friends: {
+			name: [
+				'Frank',
+				'Grace'
+			]
+		}
+	}
+]
+```
+
+### Flattening Nested Arrays
+
+If `friends` themselves have a `friends` array, creating nested arrays:
+
+
+```surql title="Create a new person record"
+CREATE person:5 CONTENT {
+    name: "Eve",
+    friends: [
+        {
+            id: "person:6",
+            name: "Frank",
+            friends: [
+                { id: "person:9", name: "Ivan" },
+                { id: "person:10", name: "Judy" }
+            ]
+        },
+        {
+            id: "person:7",
+            name: "Grace",
+            friends: [
+                { id: "person:11", name: "Karl" }
+            ]
+        }
+    ]
+};
+```
+
+```json title="Response"
+{
+  "id": "person:5",
+  "name": "Eve",
+  "friends": [
+    {
+      "id": "person:6",
+      "name": "Frank",
+      "friends": [
+        { "id": "person:9", "name": "Ivan" },
+        { "id": "person:10", "name": "Judy" }
+      ]
+    },
+    {
+      "id": "person:7",
+      "name": "Grace",
+      "friends": [
+        { "id": "person:11", "name": "Karl" }
+      ]
+    }
+  ]
+}
+```
+
+To get a flat list of all friends' friends:
+
+```surql title="Get a flat list of all friends' friends"
+SELECT friends.friends... FROM person WHERE id = r'person:5';
+```
+
+```json title="Response"
+[
+  {
+    "friends": [
+      { "id": "person:9", "name": "Ivan" },
+      { "id": "person:10", "name": "Judy" },
+      { "id": "person:11", "name": "Karl" }
+    ]
+  }
+]
+```
+
+
+## Notes on Idioms
+
+- **Chaining**: Idioms can be chained to traverse deeply nested structures.
+- **Performance**: Be mindful of performance when using complex idioms; indexing fields can help.
```

Comment:
This point is the same as Index Fields below. I feel like these two sections could be combined into a final "Notes and best practices on idioms"  type section.